DependencyInjection/DoctrineExtension.php

```diff
@@ -132,9 +132,16 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             $logger = new Reference('doctrine.dbal.logger');
         }
         unset($connection['logging']);
-        if ($connection['profiling']) {
-            $profilingLoggerId = 'doctrine.dbal.logger.profiling.' . $name;
-            $container->setDefinition($profilingLoggerId, new ChildDefinition('doctrine.dbal.logger.profiling'));
+
+        if ($connection['profiling'] || $connection['profiling_backtrace']) {
+            if ($connection['profiling_backtrace']) {
+                $profilingAbstractId = 'doctrine.dbal.logger.backtrace';
+            } else {
+                $profilingAbstractId = 'doctrine.dbal.logger.profiling';
+            }
```

Comment:
Maybe us a ternary here
```php
<?php
$profilingAbstractId = $connection['profiling_backtrace'] ?
    'doctrine.dbal.logger.backtrace' : 
    'doctrine.dbal.logger.profiling';
```

I did not mutualize the prefix to keep things greppable

Reply:
done